src/smolagents/models.py

```diff
@@ -768,8 +772,13 @@ def __call__(
             model=self.model_id,
             custom_role_conversions=self.custom_role_conversions,
             convert_images_to_image_urls=True,
+            **self.chat_completion_kwargs,
             **kwargs,
         )
+
+        if "o1" in self.model_id and "max_tokens" in completion_kwargs:
```

Comment:
Without this it would complain that only `max_completion_tokens` is supported and `max_tokens` isn't for this model. Which `smolagents` sets by default somewhere AFAICT.